connector_amazon/models/amazon_backend.py

```diff
@@ -144,3 +168,271 @@ def import_report(self):
+        """ country is mandatory, not state
+        """
+        country_code = part_ship.get('country')
+        state_name = part_ship.get('state')
+        country = self.env['res.country'].search([('code', '=', country_code)])
+        if not country or not country_code:
+            raise UserError(
+                _("Unknow country code %s in sale %s ") % (
+                    country_code, origin))
+        state = False
+        if state_name:
+            state = self.env['res.country.state'].search(
+                [('name', '=', state_name)])
+            if not state:
+                raise UserError(
+                    _("Unknown state code %s in sale %s ") % (
+                        state_name, origin))
+        return(country.id, getattr(state, 'id', state))
+
+    @api.multi
+    def import_fba_delivered_sales(self):
+        """ Import from Fulfillment by Amazon
+        """
+        for record in self:
+            mws = record._get_connection()
+            start = fields.Datetime.from_string(self.import_fba_from)
+            try:
+                sales = mws.list_orders(
+                    CreatedAfter=start.isoformat(), OrderStatus=['Shipped'],
+                    # marketplace must be in a list: weird Amazon !
+                    MarketplaceId=[record.marketplace])
+                _logger.info('%s FBA amazon sales will be imported',
+                             len(sales.ListOrdersResult.Orders.Order))
+                for order in sales.ListOrdersResult.Orders.Order:
+                    _logger.debug(order)
+                    data = self._import_fba_sale(mws, order)
+                    if self.env[('sale.order')].search([
+                            ('origin', '=', data['auto_insert']['origin'])]):
+                        continue
+                    sale_date = data['auto_insert']['date_order']
+                    self._create_sale(data)
+                    record.import_fba_from = iso8601.parse_date(sale_date)
+                    # We commit to avoid than a fail sale import
+                    # prevent to save other valid sales
+                    self._cr.commit()
+                    # prevent to be throttled by Amazon
+                    time.sleep(record.elapsed_time)
+            except BotoServerError as bs:
+                # pass
+                message = _('Amazon BotoServerError %s %s %s') % (
+                    bs.status, bs.reason, bs.body)
+                # TODO I need to commit message_post() action
+                # but older actions since last commit: how to do it ?
+                self.message_post(body=message, subtype='mail.mt_comment')
+                raise UserError(message)
+            except Exception as e:
+                message = "Amazon exception '%s'" % e.message
+                self.message_post(body=message, subtype='mail.mt_comment')
+                raise UserError(e.message)
+
+    @api.multi
+    def _import_fba_sale(self, mws, order):
```

Comment:
As this method do not import the fba_sale but etract the data from it, maybe we should call it "_extract_fba_sale" ?